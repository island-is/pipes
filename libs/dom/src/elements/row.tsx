import type { AnyElement, JSX, SpecifixJSX } from "./jsx.js";

export type IRow = SpecifixJSX<"Row", null, AnyElement | AnyElement[] | null | string>;
export const Row = (props: Omit<IRow, "type" | "children">, children: JSX | JSX[] | null | string): IRow => {
  return {
    type: "Row",
    ...props,
    children,
  };
};
