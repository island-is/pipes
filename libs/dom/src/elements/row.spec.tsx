import { describe } from "node:test";

import React from "react";

import { Row } from "./row.js";
import { testJSX } from "./snap-test.js";

const simpleRow = (number: number) => {
  return ["One row", "Two row", "Three row", "Four row", "Five row", "Six rows"].slice(0, number);
};

const bigRow = (number: number) => {
  return Array(number)
    .fill("1")
    .map((e, index) => `${index + 1}`.repeat(200));
};

describe("Render row", () => {
  describe("ansi", async () => {
    const tests = [
      testJSX(<Row>One row</Row>, "One row", import.meta.url),
      testJSX(<Row>{simpleRow(2)}</Row>, simpleRow(2).slice(-1)[0], import.meta.url),
      testJSX(<Row>{simpleRow(3)}</Row>, simpleRow(3).slice(-1)[0], import.meta.url),
      testJSX(<Row>{simpleRow(4)}</Row>, simpleRow(4).slice(-1)[0], import.meta.url),
      testJSX(<Row>{simpleRow(5)}</Row>, simpleRow(5).slice(-1)[0], import.meta.url),
      testJSX(<Row>{simpleRow(6)}</Row>, simpleRow(6).slice(-1)[0], import.meta.url),
      testJSX(<Row>{bigRow(1)}</Row>, "1", import.meta.url),
      testJSX(<Row>{bigRow(2)}</Row>, "2", import.meta.url),
      testJSX(<Row>{bigRow(3)}</Row>, "3", import.meta.url),
      testJSX(<Row>{bigRow(4)}</Row>, "4", import.meta.url),
      testJSX(<Row>{bigRow(5)}</Row>, "5", import.meta.url),
      testJSX(<Row>{bigRow(6)}</Row>, "6", import.meta.url),
    ];
    await Promise.all(tests);
  });
});
