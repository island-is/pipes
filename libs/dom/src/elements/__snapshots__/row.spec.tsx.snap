// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌──────────────────────────────────────────────────────────────────────────────┐[39m[22m
[2m[90m│[39m[22m  11111111111111111111111111111111111111111111111111111111111111111111111111  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111111111111111111111111111111111111111111111111111111111111111  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111111111111111111111111111111111111111111111                        [2m[90m│[39m[22m
[2m[90m└──────────────────────────────────────────────────────────────────────────────┘[39m[22m"
`;

exports[`2-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌──────────────────────────────────────┐┌──────────────────────────────────────┐[39m[22m
[2m[90m│[39m[22m  1111111111111111111111111111111111  [2m[90m││[39m[22m  2222222222222222222222222222222222  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111111111111111111111111111  [2m[90m││[39m[22m  2222222222222222222222222222222222  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111111111111111111111111111  [2m[90m││[39m[22m  2222222222222222222222222222222222  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111111111111111111111111111  [2m[90m││[39m[22m  2222222222222222222222222222222222  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111111111111111111111111111  [2m[90m││[39m[22m  2222222222222222222222222222222222  [2m[90m│[39m[22m
[2m[90m│[39m[22m  111111111111111111111111111111      [2m[90m││[39m[22m  222222222222222222222222222222      [2m[90m│[39m[22m
[2m[90m└──────────────────────────────────────┘└──────────────────────────────────────┘[39m[22m"
`;

exports[`3-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌────────────────────────┐┌────────────────────────┐┌────────────────────────┐[39m[22m
[2m[90m│[39m[22m  11111111111111111111  [2m[90m││[39m[22m  22222222222222222222  [2m[90m││[39m[22m  33333333333333333333  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111111111  [2m[90m││[39m[22m  22222222222222222222  [2m[90m││[39m[22m  33333333333333333333  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111111111  [2m[90m││[39m[22m  22222222222222222222  [2m[90m││[39m[22m  33333333333333333333  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111111111  [2m[90m││[39m[22m  22222222222222222222  [2m[90m││[39m[22m  33333333333333333333  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111111111  [2m[90m││[39m[22m  22222222222222222222  [2m[90m││[39m[22m  33333333333333333333  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111111111  [2m[90m││[39m[22m  22222222222222222222  [2m[90m││[39m[22m  33333333333333333333  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111111111  [2m[90m││[39m[22m  22222222222222222222  [2m[90m││[39m[22m  33333333333333333333  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111111111  [2m[90m││[39m[22m  22222222222222222222  [2m[90m││[39m[22m  33333333333333333333  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111111111  [2m[90m││[39m[22m  22222222222222222222  [2m[90m││[39m[22m  33333333333333333333  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111111111  [2m[90m││[39m[22m  22222222222222222222  [2m[90m││[39m[22m  33333333333333333333  [2m[90m│[39m[22m
[2m[90m└────────────────────────┘└────────────────────────┘└────────────────────────┘[39m[22m"
`;

exports[`4-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌──────────────────┐┌──────────────────┐┌──────────────────┐┌──────────────────┐[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  11111111111111  [2m[90m││[39m[22m  22222222222222  [2m[90m││[39m[22m  33333333333333  [2m[90m││[39m[22m  44444444444444  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111            [2m[90m││[39m[22m  2222            [2m[90m││[39m[22m  3333            [2m[90m││[39m[22m  4444            [2m[90m│[39m[22m
[2m[90m└──────────────────┘└──────────────────┘└──────────────────┘└──────────────────┘[39m[22m"
`;

exports[`5-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌──────────────┐┌──────────────┐┌──────────────┐┌──────────────┐┌──────────────┐[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111111  [2m[90m││[39m[22m  2222222222  [2m[90m││[39m[22m  3333333333  [2m[90m││[39m[22m  4444444444  [2m[90m││[39m[22m  5555555555  [2m[90m│[39m[22m
[2m[90m└──────────────┘└──────────────┘└──────────────┘└──────────────┘└──────────────┘[39m[22m"
`;

exports[`6-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌───────────┐┌───────────┐┌───────────┐┌───────────┐┌───────────┐┌───────────┐[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111111  [2m[90m││[39m[22m  2222222  [2m[90m││[39m[22m  3333333  [2m[90m││[39m[22m  4444444  [2m[90m││[39m[22m  5555555  [2m[90m││[39m[22m  6666666  [2m[90m│[39m[22m
[2m[90m│[39m[22m  1111     [2m[90m││[39m[22m  2222     [2m[90m││[39m[22m  3333     [2m[90m││[39m[22m  4444     [2m[90m││[39m[22m  5555     [2m[90m││[39m[22m  6666     [2m[90m│[39m[22m
[2m[90m└───────────┘└───────────┘└───────────┘└───────────┘└───────────┘└───────────┘[39m[22m"
`;

exports[`Five big rows-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `""`;

exports[`Five row-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌──────────────┐┌──────────────┐┌──────────────┐┌──────────────┐┌──────────────┐[39m[22m
[2m[90m│[39m[22m  One row     [2m[90m││[39m[22m  Two row     [2m[90m││[39m[22m  Three row   [2m[90m││[39m[22m  Four row    [2m[90m││[39m[22m  Five row    [2m[90m│[39m[22m
[2m[90m└──────────────┘└──────────────┘└──────────────┘└──────────────┘└──────────────┘[39m[22m"
`;

exports[`Five rows-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `""`;

exports[`Four big rows-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `""`;

exports[`Four row-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌──────────────────┐┌──────────────────┐┌──────────────────┐┌──────────────────┐[39m[22m
[2m[90m│[39m[22m  One row         [2m[90m││[39m[22m  Two row         [2m[90m││[39m[22m  Three row       [2m[90m││[39m[22m  Four row        [2m[90m│[39m[22m
[2m[90m└──────────────────┘└──────────────────┘└──────────────────┘└──────────────────┘[39m[22m"
`;

exports[`Four rows-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `""`;

exports[`One row-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌──────────────────────────────────────────────────────────────────────────────┐[39m[22m
[2m[90m│[39m[22m  One row                                                                     [2m[90m│[39m[22m
[2m[90m└──────────────────────────────────────────────────────────────────────────────┘[39m[22m"
`;

exports[`Six rows-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌───────────┐┌───────────┐┌───────────┐┌───────────┐┌───────────┐┌───────────┐[39m[22m
[2m[90m│[39m[22m  One row  [2m[90m││[39m[22m  Two row  [2m[90m││[39m[22m  Three    [2m[90m││[39m[22m  Four     [2m[90m││[39m[22m  Five     [2m[90m││[39m[22m  Six      [2m[90m│[39m[22m
[2m[90m│[39m[22m           [2m[90m││[39m[22m           [2m[90m││[39m[22m  row      [2m[90m││[39m[22m  row      [2m[90m││[39m[22m  row      [2m[90m││[39m[22m  rows     [2m[90m│[39m[22m
[2m[90m└───────────┘└───────────┘└───────────┘└───────────┘└───────────┘└───────────┘[39m[22m"
`;

exports[`Three big rows-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `""`;

exports[`Three row-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌────────────────────────┐┌────────────────────────┐┌────────────────────────┐[39m[22m
[2m[90m│[39m[22m  One row               [2m[90m││[39m[22m  Two row               [2m[90m││[39m[22m  Three row             [2m[90m│[39m[22m
[2m[90m└────────────────────────┘└────────────────────────┘└────────────────────────┘[39m[22m"
`;

exports[`Three rows-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `""`;

exports[`Two big rows-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `""`;

exports[`Two row-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `
"[2m[90m┌──────────────────────────────────────┐┌──────────────────────────────────────┐[39m[22m
[2m[90m│[39m[22m  One row                             [2m[90m││[39m[22m  Two row                             [2m[90m│[39m[22m
[2m[90m└──────────────────────────────────────┘└──────────────────────────────────────┘[39m[22m"
`;

exports[`Two rows-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `""`;

exports[`render five elements-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `"One row                Two row                Three row              Four row               Five row               "`;

exports[`render four elements-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `"One row                      Two row                      Three row                    Four row                     "`;

exports[`render four row multiple lines-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `"1111111111111111111111111111 2222222222222222222222222222 3333333333333333333333333333 4444444444444444444444444444 1111111111111111111111111111 2222222222222222222222222222 3333333333333333333333333333 4444444444444444444444444444 1111111111111111111111111111 2222222222222222222222222222 3333333333333333333333333333 4444444444444444444444444444 1111111111111111111111111111 2222222222222222222222222222 3333333333333333333333333333 4444444444444444444444444444 1111111111111111111111111111 2222222222222222222222222222 3333333333333333333333333333 4444444444444444444444444444 1111111111111111111111111111 2222222222222222222222222222 3333333333333333333333333333 4444444444444444444444444444 1111111111111111111111111111 2222222222222222222222222222 3333333333333333333333333333 4444444444444444444444444444 1111                         2222                         3333                         4444                         "`;

exports[`render one element-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `"One row                                                                                                            "`;

exports[`render three elements-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `"One row                               Two row                               Three row                             "`;

exports[`render three five multiple lines-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `"1111111111111111111111 2222222222222222222222 3333333333333333333333 4444444444444444444444 5555555555555555555555 1111111111111111111111 2222222222222222222222 3333333333333333333333 4444444444444444444444 5555555555555555555555 1111111111111111111111 2222222222222222222222 3333333333333333333333 4444444444444444444444 5555555555555555555555 1111111111111111111111 2222222222222222222222 3333333333333333333333 4444444444444444444444 5555555555555555555555 1111111111111111111111 2222222222222222222222 3333333333333333333333 4444444444444444444444 5555555555555555555555 1111111111111111111111 2222222222222222222222 3333333333333333333333 4444444444444444444444 5555555555555555555555 1111111111111111111111 2222222222222222222222 3333333333333333333333 4444444444444444444444 5555555555555555555555 1111111111111111111111 2222222222222222222222 3333333333333333333333 4444444444444444444444 5555555555555555555555 1111111111111111111111 2222222222222222222222 3333333333333333333333 4444444444444444444444 5555555555555555555555 11                     22                     33                     44                     55                     "`;

exports[`render three row multiple lines-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `"1111111111111111111111111111111111111 2222222222222222222222222222222222222 3333333333333333333333333333333333333 1111111111111111111111111111111111111 2222222222222222222222222222222222222 3333333333333333333333333333333333333 1111111111111111111111111111111111111 2222222222222222222222222222222222222 3333333333333333333333333333333333333 1111111111111111111111111111111111111 2222222222222222222222222222222222222 3333333333333333333333333333333333333 1111111111111111111111111111111111111 2222222222222222222222222222222222222 3333333333333333333333333333333333333 111111111111111                       222222222222222                       333333333333333                       "`;

exports[`render two elements-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `"One row                                                  Two row                                                  "`;

exports[`render two row multiple lines-file:///var/home/lommi/Projects/island.is-pipes/libs/dom/src/elements/row.spec.tsx 1`] = `"11111111111111111111111111111111111111111111111111111111 22222222222222222222222222222222222222222222222222222222 11111111111111111111111111111111111111111111111111111111 22222222222222222222222222222222222222222222222222222222 11111111111111111111111111111111111111111111111111111111 22222222222222222222222222222222222222222222222222222222 11111111111111111111111111111111                         22222222222222222222222222222222                         "`;
